### PR TITLE
Refactor: project card and group district options

### PR DESCRIPTION
**Pull request by millaeerola**
- project card and group district options now come from a separate table, so we can also have them for projects and groups that don't have them in the hierarchy

### CODE REVIEW

File: src/forms/useProjectForm.ts [hunk 1]
```diff
@@ -73,28 +68,24 @@ const useProjectFormValues = () => {
...
+    const selectedDivision = getSelectedLocation(divisions, selectedSubDivision?.parent, project?.projectDistrict);
 
-    const selectedDivision = projectLocationId
-      ? divisions.find(({ id }) => id === projectLocationId)
-      : undefined;
-
-    const districtId = selectedDivision?.parent ?? project?.projectLocation;
-
-    const selectedDistrict = districtId ? districts.find(({ id }) => id === districtId) : undefined;
+    const selectedDistrict = getSelectedLocation(districts, selectedDivision?.parent, project?.projectDistrict);
```
Comment (JenniLehtonen) [inline src/forms/useProjectForm.ts:88]: The projectDistrict could be named differently

File: src/components/Planning/GroupDialog/GroupDialog.tsx [hunk 2]
```diff
@@ -45,19 +51,32 @@ const buildRequestPayload = (
   return data;
 };
 
+const hierarchyDivisionsAsIoptions = (districtName: string, subClassId: string, classId: string, hierarchyDistricts: ILocation[], hierarchyDivisions: ILocation[]) => {
+  const relatedDistricts = hierarchyDistricts.filter(({ parentClass }) => parentClass === subClassId || classId);
+  const relatedDistrict = relatedDistricts.find(({ name }) => name.includes(districtName));
```
[the remark this answers is not in the document]
Comment (millaeerola) [reply src/components/Planning/GroupDialog/GroupDialog.tsx:56]: The intention is to first compare the parentClass to subClassId and classId. I think using parentClass === (subClassId || classId) instead would only compare the parentClass to subClassId if that is not undefined and leave out the classId completely from the comparison in that case

File: src/utils/common.ts [hunk 3]
```diff
@@ -88,43 +101,114 @@ const syncConstructionDates = (request: IProjectRequest, form: IAppForms) => {
...
+  return location;
+}
+
+const getLocationList = (list: ILocation[] | undefined, parentId?: string, pClass?: string) => {
+  let locationList: ILocation[] = [];
+  if (pClass && list) {
+    locationList = list.filter(({ parentClass }) => parentClass === pClass);
+  } 
+  else if (parentId && list) {
+    locationList = list.filter(({ parent }) => parent === parentId);
```
[the remark this answers is not in the document]
Comment (millaeerola) [reply src/utils/common.ts]: I wrote some comments. Could you check if they make any sense to you?

[the remark this answers is not in the document]
Comment (JenniLehtonen) [reply src/utils/common.ts]: Looks good!

**Review by markspl — APPROVED** [review]
👍

**Pull request merged**